src/error.rs

```diff
@@ -3,105 +3,129 @@
 use std::io;
 
 use std::path::PathBuf;
-use std::error;
 use std::sync::PoisonError;
-use directory::error::{OpenReadError, OpenWriteError, OpenDirectoryError};
+use directory::error::{IOError, OpenReadError, OpenWriteError, OpenDirectoryError};
 use query;
 use schema;
 use fastfield::FastFieldNotAvailableError;
 use serde_json;
 
-
-/// Generic tantivy error.
-///
-/// Any specialized error return in tantivy can be converted in `tantivy::Error`.
-#[derive(Debug)]
-pub enum Error {
-    /// Path does not exist.
-    PathDoesNotExist(PathBuf),
-    /// File already exists, this is a problem when we try to write into a new file.
-    FileAlreadyExists(PathBuf),
-    /// IO Error
-    IOError(io::Error),
-    /// A thread holding the locked panicked and poisoned the lock.
-    Poisoned,
-    /// The data within is corrupted.
-    ///
-    /// For instance, it contains invalid JSON.
-    CorruptedFile(PathBuf, Box<error::Error + Send + Sync>),
-    /// Invalid argument was passed by the user.
-    InvalidArgument(String),
-    /// An Error happened in one of the thread
-    ErrorInThread(String),
-    /// An Error appeared related to the lack of a field.
-    SchemaError(String),
-    /// Tried to access a fastfield reader for a field not configured accordingly.
-    FastFieldError(FastFieldNotAvailableError),
-}
+error_chain!(
+    errors {
+        PathDoesNotExist(buf: PathBuf) {
+            description("path does not exist")
```

Comment:
have you checked that the rustdoc looked ok? On the first PR, the doc was not generated for the error kind